js/app.js

```diff
@@ -81,20 +68,98 @@ TextApp.prototype.setTheme = function() {
   this.editor_.setTheme(theme);
 };
 
+/**
+ * Remove the editor so it can be reinitialized.
+ * @param editor The DOM element containing the editor.
+ */
+TextApp.prototype.removeEditor = function(editor) {
+  // Let the object do any clean up it needs.
+  if (this.editor_ !== null) {
+    this.editor_.destory();
+  }
+
+  // Clear the DOM
+  while (editor.firstElementChild !== null) {
+    editor.firstElementChild.remove();
+  }
+};
+
 /**
  * Called when all the services have started and settings are loaded.
  */
 TextApp.prototype.onSettingsReady_ = function() {
+  this.settingsController_ = new SettingsController(this.settings_);
+
+  this.initEditor_();
+
+  this.analytics_.setEnabled(this.settings_.get('analytics'));
+  this.analytics_.reportSettings(this.settings_);
+  this.windowController_.setAlwaysOnTop(this.settings_.get('alwaysontop'));
+
+  chrome.runtime.getBackgroundPage(function(bg) {
+    bg.background.onWindowReady(this);
+  }.bind(this));
+};
+
+/**
+ * Create a new editor and load all settings.
+ */
+TextApp.prototype.initEditor_ = function() {
+  // Remove any editor that already exists.
+  if (this.editor_ !== null) {
+    const editor = document.getElementById('editor');
+    this.removeEditor(editor);
+  }
+
+  if (this.settings_.get('screenreadermode')) {
+    this.editor_ = new EditorTextArea(editor, this.settings_);
+  } else {
+    this.editor_ = new EditorCodeMirror(editor, this.settings_);
+  }
+
+  if (!this.tabs_) {
+    // If tabs doesn't exist this is the first editor being created, if so
```

Comment:
optional: this function is pretty big, consider pulling some of this logic out into a separate function.

Reply:
Done!